### ISSUE TITLE
May I suggest some discipline from the maintainers

### ISSUE THREAD

**Issue by tthsqe12** [official-stockfish/Stockfish#1207]
Here is proof that "thread code reformat" was a functional change (possibly bad).
I downloaded the linux modern 64 builds from august 12 and august 13 versions from abroke.
If these binaries are not reliable, then please take this message with a grain of salt. However, I think these is something here.

for clarity

stockfish_17081211 =
```
Author: tthsqe12 
Date: Sat Aug 12 02:47:11 2017 -0700 
Timestamp: 1502531231 

Fix the handling of opposite bishops in KXK endgame evaluation 
```
stockfish_17081313 =
```
Author: Marco Costalba 
Date: Sun Aug 13 04:41:59 2017 -0700 
Timestamp: 1502624519 

Thread code reformat 

Simplify out low level sync stuff (mutex 
and friends) and avoid to use them directly 
in many functions. 

Also some renaming and better comment while 
there. 

No functional change. 
```

The claim of "No functional change" is wrong. Functional changes should be tested, and in this case I am almost certain this patch would fail.

```
$ ./stockfish_17081313
Stockfish 130817 64 POPCNT by T. Romstad, M. Costalba, J. Kiiski, G. Linscott
position fen r1bqkb1r/pppppppp/n4n2/8/2PP4/8/PP2PPPP/RNBQKBNR w KQkq - 0 1 moves b1c3 e7e6 e2e4 d7d5 e4e5 f6e4 g1e2 e4c3 e2c3 c7c5 c4d5 c5d4 f1a6 d4c3 a6b5 c8d7 d5e6 f7e6 d1d7 d8d7 b5d7 e8d7 b2c3 a8c8 e1e2 c8c3 c1e3 f8c5 a1d1 d7e7 d1c1 c3c1 h1c1 c5e3 e2e3 h8d8 g2g4 b7b6 h2h4 d8d5 c1c7 d5d7 c7c8 d7d5 c8c7
go depth 9
info depth 1 seldepth 1 multipv 1 score cp -124 nodes 24 nps 12000 tbhits 0 time 2 pv d5d7
info depth 2 seldepth 3 multipv 1 score cp -27 nodes 40 nps 20000 tbhits 0 time 2 pv d5d7 c7d7 e7d7
info depth 3 seldepth 4 multipv 1 score cp -34 nodes 92 nps 46000 tbhits 0 time 2 pv d5d7 c7c1 a7a5
info depth 4 seldepth 5 multipv 1 score cp -35 nodes 194 nps 97000 tbhits 0 time 2 pv d5d7 c7d7 e7d7 f2f4 g7g6
info depth 5 seldepth 6 multipv 1 score cp -52 nodes 688 nps 229333 tbhits 0 time 3 pv d5d7 c7c6 b6b5 f2f4 e7f7
info depth 6 seldepth 6 multipv 1 score cp -45 nodes 927 nps 309000 tbhits 0 time 3 pv d5d7 c7c6 b6b5 f2f4 g7g6 c6a6
info depth 7 seldepth 7 multipv 1 score cp -49 nodes 1615 nps 403750 tbhits 0 time 4 pv d5d7 c7c6 b6b5 f2f4 b5b4 c6a6 e7f7
info depth 8 seldepth 9 multipv 1 score cp -49 nodes 4221 nps 603000 tbhits 0 time 7 pv d5d7 c7c8 d7d8 c8c6 e7d7 c6c1 g7g6 c1d1 d7e7
info depth 9 seldepth 12 multipv 1 score cp -32 nodes 22559 nps 902360 tbhits 0 time 25 pv d5d7 c7d7 e7d7 f2f4 g7g6 e3e4 d7c6 a2a4 c6d7 f4f5 g6f5 g4f5
bestmove d5d7 ponder c7d7
quit
$ ./stockfish_17081211
Stockfish 120817 64 POPCNT by T. Romstad, M. Costalba, J. Kiiski, G. Linscott
position fen r1bqkb1r/pppppppp/n4n2/8/2PP4/8/PP2PPPP/RNBQKBNR w KQkq - 0 1 moves b1c3 e7e6 e2e4 d7d5 e4e5 f6e4 g1e2 e4c3 e2c3 c7c5 c4d5 c5d4 f1a6 d4c3 a6b5 c8d7 d5e6 f7e6 d1d7 d8d7 b5d7 e8d7 b2c3 a8c8 e1e2 c8c3 c1e3 f8c5 a1d1 d7e7 d1c1 c3c1 h1c1 c5e3 e2e3 h8d8 g2g4 b7b6 h2h4 d8d5 c1c7 d5d7 c7c8 d7d5 c8c7
go depth 9
info depth 1 seldepth 1 multipv 1 score cp -124 nodes 24 nps 12000 tbhits 0 time 2 pv d5d7
info depth 2 seldepth 3 multipv 1 score cp -27 nodes 40 nps 20000 tbhits 0 time 2 pv d5d7 c7d7 e7d7
info depth 3 seldepth 4 multipv 1 score cp -34 nodes 92 nps 46000 tbhits 0 time 2 pv d5d7 c7c1 a7a5
info depth 4 seldepth 5 multipv 1 score cp -35 nodes 194 nps 97000 tbhits 0 time 2 pv d5d7 c7d7 e7d7 f2f4 g7g6
info depth 5 seldepth 6 multipv 1 score cp -52 nodes 688 nps 344000 tbhits 0 time 2 pv d5d7 c7c6 b6b5 f2f4 e7f7
info depth 6 seldepth 6 multipv 1 score cp -45 nodes 927 nps 309000 tbhits 0 time 3 pv d5d7 c7c6 b6b5 f2f4 g7g6 c6a6
info depth 7 seldepth 7 multipv 1 score cp -49 nodes 1615 nps 403750 tbhits 0 time 4 pv d5d7 c7c6 b6b5 f2f4 b5b4 c6a6 e7f7
info depth 8 seldepth 9 multipv 1 score cp -49 nodes 4221 nps 703500 tbhits 0 time 6 pv d5d7 c7c8 d7d8 c8c6 e7d7 c6c1 g7g6 c1d1 d7e7
info depth 9 seldepth 12 multipv 1 score cp -40 nodes 16934 nps 996117 tbhits 0 time 17 pv d5d7 c7c8 d7d8 c8d8 e7d8 h4h5 d8e7 f2f4 g7g6
bestmove d5d7 ponder c7c8
quit
$ 
```

**Comment by mcostalba** [comment]
Non functional test it means bench is the same. 

In your case you tested with setting some moves after setting the position triggering a bug (that I have already identified) I will push a fix in few hours.

For the future I will be more careful and will run a regression testing in case of patches that exercise code paths not triggered by a simple bench.

I am also thinking to improve bench to detect such cases.

**Comment by tthsqe12** [comment]
Alright - I see that that push actually had two commits in it. The former looks more suspicious than the latter.

**Comment by lantonov** [comment]
(Copied from https://github.com/lantonov/asmFish/issues/77)
When trying to compile Stockfish-bdeda52 the compiler (gcc 7.1.0) gives an error and fails to compile:
```
$ mingw32-make profile-build ARCH=x86-64-bmi2 COMP=mingw

Config:
debug: 'no'
sanitize: 'no'
optimize: 'yes'
arch: 'x86_64'
bits: '64'
kernel: 'MINGW64_NT-10.0'
os: 'Windows_NT'
prefetch: 'yes'
popcnt: 'yes'
sse: 'yes'
pext: 'yes'

Flags:
CXX: g++
CXXFLAGS: -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11  -Wextra -Wshad                                                                                                                                  ow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_PEXT -mbmi2
LDFLAGS:  -static

Testing config sanity. If this fails, try 'make help' ...


Step 1/4. Building instrumented executable ...
C:/msys2/mingw64/bin/mingw32-make ARCH=x86-64-bmi2 COMP=mingw gcc-profile-make
mingw32-make[1]: Entering directory 'C:/msys2/home/lanto/src'
C:/msys2/mingw64/bin/mingw32-make ARCH=x86-64-bmi2 COMP=mingw \
EXTRACXXFLAGS='-fprofile-generate' \
EXTRALDFLAGS='-lgcov' \
all
mingw32-make[2]: Entering directory 'C:/msys2/home/lanto/src'
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o benchmark.o benchmark.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o bitbase.o bitbase.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o bitboard.o bitboard.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o endgame.o endgame.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o evaluate.o evaluate.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o main.o main.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o material.o material.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o misc.o misc.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o movegen.o movegen.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o movepick.o movepick.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o pawns.o pawns.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o position.o position.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o psqt.o psqt.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o search.o search.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o thread.o thread.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o timeman.o timeman.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o tt.o tt.cpp
g++ -Wall -Wcast-qual -fno-exceptions -fno-rtti -std=c++11 -fprofile-generate -W                                                                                                                                  extra -Wshadow -DNDEBUG -O3 -DIS_64BIT -msse -msse3 -mpopcnt -DUSE_POPCNT -DUSE_                                                                                                                                  PEXT -mbmi2   -c -o uci.o uci.cpp
In file included from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/x86_64-w64-mingw32/bits/c++allocator.h:33:0,
                 from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/bits/allocator.h:46,
                 from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/string:41,
                 from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/bits/locale_classes.h:40,
                 from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/bits/ios_base.h:41,
                 from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/ios:42,
                 from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/ostream:38,
                 from C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/                                                                                                                                  c++/iostream:39,
                 from uci.cpp:22:
C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/c++/ext/new_allocator.                                                                                                                                  h: In instantiation of 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Arg                                                                                                                                  s&& ...) [with _Up = Thread; _Args = {int}; _Tp = Thread]':
C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/c++/bits/alloc_traits.                                                                                                                                  h:475:4:   required from 'static void std::allocator_traits<std::allocator<_Char                                                                                                                                  T> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&,                                                                                                                                   _Up*, _Args&& ...) [with _Up = Thread; _Args = {int}; _Tp = Thread; std::alloca                                                                                                                                  tor_traits<std::allocator<_CharT> >::allocator_type = std::allocator<Thread>]'
C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/c++/bits/shared_ptr_ba                                                                                                                                  se.h:1325:23:   required from 'std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_S                                                                                                                                  p_make_shared_tag, const _Alloc&, _Args&& ...) [with _Alloc = std::allocator<Thr                                                                                                                                  ead>; _Args = {int}; _Tp = Thread; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lo                                                                                                                                  ck_policy)2]'
C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/c++/bits/shared_ptr.h:                                                                                                                                  344:64:   required from 'std::shared_ptr<_Tp>::shared_ptr(std::_Sp_make_shared_t                                                                                                                                  ag, const _Alloc&, _Args&& ...) [with _Alloc = std::allocator<Thread>; _Args = {                                                                                                                                  int}; _Tp = Thread]'
C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/c++/bits/shared_ptr.h:                                                                                                                                  690:14:   required from 'std::shared_ptr<_Tp> std::allocate_shared(const _Alloc&                                                                                                                                  , _Args&& ...) [with _Tp = Thread; _Alloc = std::allocator<Thread>; _Args = {int                                                                                                                                  }]'
C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/c++/bits/shared_ptr.h:                                                                                                                                  706:39:   required from 'std::shared_ptr<_Tp> std::make_shared(_Args&& ...) [wit                                                                                                                                  h _Tp = Thread; _Args = {int}]'
uci.cpp:151:45:   required from here
C:/msys2/mingw64/lib/gcc/x86_64-w64-mingw32/7.1.0/include/c++/ext/new_allocator.                                                                                                                                  h:136:4: error: no matching function for call to 'Thread::Thread(int)'
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from uci.cpp:30:0:
thread.h:51:3: note: candidate: Thread::Thread()
   Thread();
   ^~~~~~
thread.h:51:3: note:   candidate expects 0 arguments, 1 provided
<builtin>: recipe for target 'uci.o' failed
mingw32-make[2]: *** [uci.o] Error 1
mingw32-make[2]: Leaving directory 'C:/msys2/home/lanto/src'
Makefile:532: recipe for target 'gcc-profile-make' failed
mingw32-make[1]: *** [gcc-profile-make] Error 2
mingw32-make[1]: Leaving directory 'C:/msys2/home/lanto/src'
Makefile:436: recipe for target 'profile-build' failed
mingw32-make: *** [profile-build] Error 2
```
The next commit, stockfish-df6cb44, compiles normally without warnings.

**Comment by mcostalba** [comment]
Fix has been merged with 232c50fed0b80a0f
